### DISCUSSION TITLE
HOGWARTS: Sorting Hat cards aren't printing properly to the DOM

### DISCUSSION THREAD

**Discussion by elliewalkersm** [nss-evening-cohort-14/evening-client-side#66] [Help Me!] [answered]
Have you searched the other help tickets? If not, do that first (Duplicates asks for help will be shamed ceaselessly)
yes

Describe your issue
I am having trouble getting my object array to print to the dom. I'm not sure which Id I need to target.

List any error messages that appear
none

What have you tried? Tell us everything you have tried.
Things I've searched:
https://github.com/elliewalkersm/event-listeners/blob/pies/main.js
https://stackoverflow.com/questions/14895287/how-to-print-object-array-in-javascript
https://stackoverflow.com/questions/53616995/loop-objects-in-array-to-create-element

Copy/paste link TO THE FILE on your branch or in your repo that you are having issues with
https://github.com/elliewalkersm/sorting-hat/blob/form/main.js

**Comment by drteresavasquez** [comment]
@elliewalkersm to help you more effectively, can you please edit your issue with these items:

- Have you searched the other help tickets? If not, do that first (Duplicates asks for help will be shamed ceaselessly)
- Describe your issue
- List any error messages that appear
- What have you tried? Tell us everything you have tried.
- Copy/paste link TO THE FILE on your branch or in your repo that you are having issues with
- NOTE: We will not accept links to the whole repo. You must provide a link directly to the specific file

That last one is important. We need the link directly to the file in question please.

**Reply by elliewalkersm** [reply]
updated

**Reply by drteresavasquez** [reply]
On the 'What have you tried' question, we need you to specify what you tried to get this working so that we can evaluate what you did and if you were on the right track 😄

**Comment by drteresavasquez** [comment]
@elliewalkersm here: https://github.com/elliewalkersm/sorting-hat/blob/form/main.js#L66-L69 your code looks like this:

```
const buttonClick = () => {
  document.querySelector('#start-btn').addEventListener('click', 'function');
  document.querySelector('#sort-btn').addEventListener('click', 'function');
}
```

What is that callback? Are your button clicks working at all?

**Reply by elliewalkersm** [reply]
I wrote that as a place holder for when I get the rest working. Right now, I'm not able to get the createCard function to display the cards from the array.

**Reply by drteresavasquez** [reply]
a couple of things:

1. ~You are calling the card builder function with 1 argument and it looks like the function is expecting 2 parameters~
2. You have an extra div tag in your HTML

**Reply by drteresavasquez** [reply]
Wait...I take 1 back. Let me look more closely at that function really quickly

**Comment by drteresavasquez** [comment, marked as answer]
@elliewalkersm so this is one of the reasons we need to see the error messages. When I run your code, I am getting this:

```
main.js:67 Uncaught TypeError: Failed to execute 'addEventListener' on 'EventTarget': parameter 2 is not of type 'Object'.
    at buttonClick (main.js:67)
    at init (main.js:72)
    at main.js:76
```

The reason for that is because you have that 'placeholder' in the lines of code I shared above. Either comment out the event listeners in your `buttonClick` function or comment out the function call in your `init` function call because your cards never render because of the error your place holder is creating.

That should get your objects printing to the DOM. Looks like the cards need a little work, but that could be an easy fix once you get them on the DOM.

**Reply by elliewalkersm** [reply]
I have commented out everything except the array, printToDom function, cardBuilder function, and the init function. I am still unable to see the cards.

**Reply by drteresavasquez** [reply]
also, To fix your cards, you just need to add a closing `"` on line 37 of your JS file. You have this:
` <a href="#" class="btn btn-danger>Expel</a>` and you need to close the quotes on the class. That should get you all fixed up!

**Reply by elliewalkersm** [reply]
That was it! Thank you!

**Reply by drteresavasquez** [reply]
@elliewalkersm were you not able to see the cards at all or was it that they were not showing properly on the DOM?
